vuu-ui/packages/vuu-filters/src/filter-clause/DateInput.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useCallback, useState } from "react";
+import { getLocalTimeZone, DateValue } from "@internationalized/date";
+import { DatePicker } from "@finos/vuu-ui-controls";
+import { toCalendarDate } from "@finos/vuu-utils";
+import { NumericFilterClauseOp } from "@finos/vuu-filter-types";
+import { FilterClauseValueEditor } from "./filterClauseTypes";
+
+interface Props
+  extends Pick<FilterClauseValueEditor<number>, "onInputComplete"> {
+  value: number | undefined;
+  operator: NumericFilterClauseOp;
+}
+
+export const DateInput: React.FC<Props> = (props) => {
+  const { value, onInputComplete, operator } = props;
+  const toEpochMilliS = getEpochMillisConverter(operator);
+
+  const [date, setDate] = useState<DateValue | undefined>(() =>
+    getInitialState(value)
+  );
+
+  const onSelectedDateChange = useCallback((d: DateValue | undefined) => {
+    setDate(d);
+  }, []);
+
+  const onBlur = useCallback(() => {
+    date && onInputComplete(toEpochMilliS(date));
+  }, [date, onInputComplete, toEpochMilliS]);
+
+  return (
+    <DatePicker
+      className={"vuuFilterClause-DatePicker"}
+      selectedDate={date}
+      onBlur={onBlur}
+      onSelectedDateChange={onSelectedDateChange}
+      closeOnSelection
+      hideOutOfRangeDates
+    />
+  );
+};
+
+function getInitialState(value: Props["value"]) {
+  return value ? toCalendarDate(new Date(value)) : undefined;
+}
+
+const getEpochMillisConverter =
```

Comment:
maybe put this in vuu-utils/date/utils ?

Reply:
I did think about it previously (either `date/utils` or `filter/utils`), but I eventually decided not to as I think that functionality is tightly coupled with filtering (specifically with the `DateInput` component itself) - it takes in an `op` and returns start of day for `=` & `!=` which isn't necessarily correct if viewed in isolation - so it won't be much of use elsewhere. Wdyt?

Reply:
fair enough. Agreed.